Debug-frontend-tests.md

```diff
@@ -78,7 +78,9 @@ describe('oppia', () => {
 });
 ```
 
-These tests will fail if `should do something` runs before `should do something else`, but not if the tests run in the reverse order. Our frontend tests run in a non-deterministic order, so these kinds of problems generally appear as flakes. In other words, these tests will sometimes pass but fail other times, even if the underlying code is the same.
+These tests will fail if `should do something` runs before `should do something else`, but not if the tests run in the reverse order. Our frontend tests run in a non-deterministic order, so these kinds of problems generally appear as intermittent failures. In other words, these tests will sometimes pass but fail other times, even if the underlying code is the same.
+
+The tests should work correctly regardless of what order they run in. If you find some tests where it matters what order the tests run in, those tests are incorrect, so you should fix them just like you would fix a failing test.
```

Comment:
Should we add a note (maybe both here + the frontend tests page) to please not write any tests that break this condition / please adhere to these condition when writing tests?

Reply:
done